core/src/mender-api.c

```diff
@@ -103,11 +103,16 @@ mender_err_t
 mender_api_init(mender_api_config_t *config) {
 
     assert(NULL != config);
-    assert(NULL != config->artifact_name);
     assert(NULL != config->device_type);
     assert(NULL != config->host);
     mender_err_t ret;
 
+    /* Load and set artifact_name here */
+    if (MENDER_OK != mender_storage_get_artifact_name(&artifact_name) && NULL != artifact_name && NULL != artifact_name->value) {
```

Comment:
Why do we need `artifact_name` to be a key-value list? We only seem to take the first value anyway and we don't care about key(s).

Reply:
Yeah, it got a bit messy since I tried to follow the way it was done in the c++ client, where we have artifact_name as a part of the provides and they delete it etc., I will rework this - and I think that only the value should be stored :+1: